src/text-input/index.ts

```diff
@@ -33,6 +33,8 @@ export interface TextInputProperties extends ThemedProperties, InputProperties,
 	placeholder?: string;
 	value?: string;
 	shouldFocus?: boolean;
+	pattern?: string | RegExp;
```

Comment:
Do you think we should a regular expression on the interface?

Reply:
I do because it's easier to catch RegExp errors in editors using RegExp syntax than in strings

Reply:
It does I guess, but then it has to be a property for the custom elements, but it would be awesome if it was just an attribute. If you keep `RegExp` on the interface, you are going to have to implement a custom diff. Because the instance's expression can be changed but it would not trigger a re-render.